sanity-io/sanity
# [base] Map legacy theme values from CSS variables to Sanity UI theme object

This change makes `@sanity/base` generate a complete `@sanity/ui` theme object based on the custom CSS properties in a given Sanity Studio. This means that the CSS properties defined by a 3rd party developer in an `override.css` will be used to generate the theme object (TS-typed by `@sanity/ui`).

### How to test

#### Test default theme

- Start the server: `npm run design-studio`
- Check that both non-`@sanity/ui` and `@sanity/ui` components use similar colors, based on the CSS properties.

#### Test custom theme

- Stop the server. Change `@import './sanity.css';` to `@import './dark.css';` in `examples/design-studio/themes/dev.css`.
- Start the server: `npm run design-studio`
- Check that both non-`@sanity/ui` and `@sanity/ui` components use similar colors, based on the CSS properties.
